packages/plugin-jira/src/plugin.ts

```diff
@@ -0,0 +1,253 @@
+      uploadResults: this.options.uploadResults ?? TRUE_VALUES.includes(process.env.ALLURE_JIRA_UPLOAD_RESULTS ?? ""),
+    };
+  }
+
+  #verifyOptions() {
+    const { token, webhook } = this.#pluginOptions;
+
+    if (!token) {
+      throw new Error(`[${this.#pluginName}] token is not set`);
+    }
+
+    if (!webhook) {
+      throw new Error(`[${this.#pluginName}] webhook is not set`);
+    }
+  }
+
+  async #requestForgeApp(props: {
+    operation: ForgeAppOperations;
+    payload: Record<string, unknown>;
+    version?: ForgeAppVersions;
+  }) {
+    this.#verifyOptions();
+    const { operation, payload, version = "v1" } = props;
+
+    const { token, webhook } = this.#pluginOptions;
+
+    try {
+      await axios.post(webhook!, {
+        operation,
+        version,
+        payload,
+        token,
+      });
+    } catch (error) {
+      if (isAxiosError(error)) {
+        throw new Error(`[${this.#pluginName}] Allure Jira Integration app error: ${error.message}`);
+      }
+      throw error;
+    }
+  }
+
+  async #uploadResults(context: PluginContext, store: AllureStore, returnData = false) {
+    this.#verifyOptions();
+
+    const statistic = await store.testsStatistic();
+
+    if (statistic.total === 0) {
+      throw new Error(`[${this.#pluginName}] no test results found`);
+    }
+
+    const testResults = prepareTestResults(await store.allTestResults());
+
+    const payload = {
+      results: testResults,
+      reportUrl: context.reportUrl,
+    };
+
+    if (returnData) {
+      return payload;
+    }
+
+    await this.#requestForgeApp({ operation: "upload-results", payload });
+  }
+
+  async #getReportStatus(store: AllureStore) {
+    const globalErrors = await store.allGlobalErrors();
+    const globalExitCode = await store.globalExitCode();
+    const code = globalExitCode?.actual ?? globalExitCode?.original;
+    const hasGlobalErrors = globalErrors.length > 0;
+
+    return hasGlobalErrors ? "failed" : code === 0 ? "passed" : "failed";
+  }
+
+  async #getStatisticByEnv(store: AllureStore) {
+    const statisticByEnv: Record<string, Statistic> = {};
+    const envs = await store.allEnvironments();
+
+    for (const env of envs) {
+      statisticByEnv[env] = await store.testsStatistic((tr) => tr.environment === env);
+    }
+
+    return statisticByEnv;
+  }
+
+  async #getCiInfo(context: PluginContext) {
+    const jobUrl = context.ci?.pullRequestUrl ?? context.ci?.jobUrl ?? context.ci?.jobRunUrl;
+    const jobLabel = context.ci?.pullRequestName ?? context.ci?.jobName ?? context.ci?.jobRunName;
+
+    return { url: jobUrl, label: jobLabel };
+  }
+
+  async #getReportDate(store: AllureStore) {
+    const testResults = await store.allTestResults();
+    return testResults.reduce((acc, { stop }) => Math.max(acc, stop || 0), 0);
+  }
+
+  async #uploadReport(context: PluginContext, store: AllureStore, returnData = false) {
+    const { reportIssue } = this.#pluginOptions;
+
+    this.#verifyOptions();
```

Comment:
Actually, we can validate options once the `done` method is called because it's the entry point of the plugin and the options can't be changed in the runtime.